posthog/api/capture.py

```diff
@@ -367,20 +371,16 @@ def get_event(request):
             # NOTE: Whilst we are testing this code we want to track exceptions but allow the events through if anything goes wrong
             capture_exception(e)
 
-        consumer_destination = "v2" if random() <= settings.REPLAY_EVENTS_NEW_CONSUMER_RATIO else "v1"
-
         try:
             replay_events, other_events = split_replay_events(events)
             processed_replay_events = replay_events
 
-            if len(replay_events) > 0:
-                # Legacy solution stays in place
+            if len(replay_events) > 0 and settings.SESSION_RECORDING_ALLOW_V1_INGESTION:
+                # Legacy solution stays in place to allow self-hosted systems a chance to upgrade
+                # or to pin their version so that they can continue to ingest recordings to ClickHouse
+                # while not receiving other fixes and upgrades
                 processed_replay_events = legacy_preprocess_session_recording_events_for_clickhouse(replay_events)
```

Comment:
I think we need to do something about the line above as this change simply doesn't preprocess...